src/components/title-list-item/title-list-item.test.js

```diff
@@ -0,0 +1,157 @@
+import {
+  screen,
+  render,
+  fireEvent,
+} from '@testing-library/react';
+
+import { Router } from 'react-router';
+import { createMemoryHistory } from 'history';
+
+import TitleListItem from './title-list-item';
+
+jest.mock('../identifiers-list', () => () => (<div>IdentifiersList component</div>));
+jest.mock('../selected-label/selected-label', () => () => (<div>SelectedLabel component</div>));
+jest.mock('../tags-label', () => () => (<div>TagsLabel component</div>));
+jest.mock('../hidden-label', () => () => (<div>HiddenLabel component</div>));
+
+const routerHistory = createMemoryHistory();
+
+const testTags = ['first', 'second', 'third'];
+
+const testPropsWithoutItem = {
+  link: 'test-link-url',
+};
+
+const testBasicProps = {
+  ...testPropsWithoutItem,
+  item: {
+    contributors: [
+      {
+        type: 'test-contributor-type',
+        contributor: 'test-contributor-name',
+      },
+    ],
+    identifiers: [
+      {
+        id: 'test-id',
+        subtype: 'test-subtype',
+        type: 'test-type',
+      },
+    ],
+    isSelected: false,
+    name: 'test-name',
+    tags: {
+      tagList: testTags,
+    },
+    visibilityData: {
+      isHidden: false,
+    },
+  },
+};
+
+const testExtendedProps = {
+  ...testBasicProps,
+  showContributors: true,
+  showIdentifiers: true,
+  showPublisherAndType: true,
+  showSelected: true,
+};
+
+const testPropsWithExtendedItem = {
+  ...testExtendedProps,
+  item: {
+    ...testExtendedProps.item,
+    publicationType: 'test-publication-type',
+    publisherName: 'test-publisher-name',
+    visibilityData: {
+      isHidden: true,
+    },
+  },
+};
+
+describe('Given TitleListItem', () => {
+  const mockOnClick = jest.fn();
+
+  const renderTitleListItem = ({ ...props }) => render(
+    <Router history={routerHistory}>
```

Comment:
You can use `<Harness>` component from `test/jest/helpers/harness`. It already has a router and other mocked components